salt/master.py

```diff
@@ -732,6 +732,13 @@ def __bind(self):
                     raise
                 # catch all other exceptions, so we don't go defunct
                 except Exception as exc:
+                    # since we are in an exceptional state, lets attempt to tell
+                    # the minion we have a problem, otherwise the minion will get
+                    # no response and be forced to wait for their max timeout
+                    try:
+                        socket.send('Unexpected Error in Mworker')
+                    except:  # pylint: disable=W0702
+                        pass
```

Comment:
As mentioned to @cachedout the wrapping except block already does log this-- as a critical log error. This specific error (unable to tell minions its busted) isn't really an error-- we just want to *try* telling the minion.